src/main/java/com/google/cloud/tools/appengine/AppEngineDescriptor.java

```diff
@@ -50,6 +51,9 @@ private AppEngineDescriptor(Document document) {
    * @throws SAXException malformed XML
    */
   public static AppEngineDescriptor parse(InputStream in) throws IOException, SAXException {
+    if (in == null) {
```

Comment:
Is this better than `Preconditions.checkNotNull()`?  Or annotating either the package or the class with `ParametersAreNonnullByDefault`?

Reply:
ParametersAreNonnullByDefault is very different.